### PR TITLE
Adding DOCTYPE parse errors

### PR DESCRIPTION
**Pull request by diervo**
/cc @inikulin

### CODE REVIEW

**Comment by inikulin** [comment]
@diervo I'll check it tomorrow if you wouldn't mind. Lot's of thing to do at the moment.

File: source [hunk 1]
```diff
@@ -103133,23 +103133,28 @@ dictionary <dfn>StorageEventInit</dfn> : <span>EventInit</span> {
...
 
    <dt>U+0027 APOSTROPHE (')</dt>
-   <dd><span>Parse error</span>. Set the DOCTYPE token's public identifier to the empty string (not
+   <dd>This is a <span data-x="parse-error-missing-whitespace-after-doctype-public-keyword">missing-whitespace-after-doctype-public-keyword</span>
+   <span>parse error</span>. Set the DOCTYPE token's public identifier to the empty string (not
    missing), then switch to the <span>DOCTYPE public identifier (single-quoted) state</span>.</dd>
 
    <dt>U+003E GREATER-THAN SIGN (&gt;)</dt>
-   <dd><span>Parse error</span>. Set the DOCTYPE token's <i data-x="force-quirks flag">force-quirks flag</i> to <i>on</i>.
+   <dd>This is an <dfn data-x="parser-error-abrupt-closing-of-doctype">abrupt-closing-of-doctype</dfn>
```
Comment (inikulin) [inline source]: It should be pars**e**-error-abrupt-closing-of-doctype. You also will need to replace `dfn` with `span` in `Before DOCTYPE public identifier state` for this error.

File: source [hunk 2]
```diff
@@ -103133,23 +103133,28 @@ dictionary <dfn>StorageEventInit</dfn> : <span>EventInit</span> {
...
 
    <dt>EOF</dt>
-   <dd><span>Parse error</span>. Set the DOCTYPE token's <i data-x="force-quirks flag">force-quirks
+   <dd>This is an <span data-x="parse-error-eof-in-doctype">eof-in-doctype</span> <span>parse error</span>.
+   Set the DOCTYPE token's <i data-x="force-quirks flag">force-quirks
    flag</i> to <i>on</i>. Emit that DOCTYPE token. Emit an end-of-file token.</dd>
 
    <dt>Anything else</dt>
-   <dd><span>Parse error</span>. Set the DOCTYPE token's <i data-x="force-quirks flag">force-quirks flag</i> to <i>on</i>.
+   <dd>This is a <dfn data-x="parse-error-missing-quote-around-doctype-public-identifier">missing-quote-around-doctype-public-identifier</dfn>
```
Comment (inikulin) [inline source]: Sorry for the wrong suggestion, we can reuse `missing-quote-before-doctype-public-identifier` error here.

File: source [hunk 3]
```diff
@@ -103229,16 +103240,19 @@ dictionary <dfn>StorageEventInit</dfn> : <span>EventInit</span> {
...
    <dt>U+003E GREATER-THAN SIGN (&gt;)</dt>
-   <dd><span>Parse error</span>. Set the DOCTYPE token's <i data-x="force-quirks flag">force-quirks flag</i> to <i>on</i>.
+   <dd>This is an <span data-x="parse-error-abrupt-closing-of-doctype">abrupt-closing-of-doctype</span>
+   <span>parse error</span>. Set the DOCTYPE token's <i data-x="force-quirks flag">force-quirks flag</i> to <i>on</i>.
    Switch to the <span>data state</span>. Emit that DOCTYPE token.</dd>
 
    <dt>EOF</dt>
-   <dd><span>Parse error</span>. Set the DOCTYPE token's <i data-x="force-quirks flag">force-quirks
-   flag</i> to <i>on</i>. Emit that DOCTYPE token. Emit an end-of-file token.</dd>
+   <dd>This is an <span data-x="parse-error-eof-in-doctype">eof-in-doctype</span><span>parse error</span>.
```
Comment (inikulin) [inline source]: Missing space before `parse error`.

File: source [hunk 4]
```diff
@@ -103264,19 +103278,23 @@ dictionary <dfn>StorageEventInit</dfn> : <span>EventInit</span> {
    <dd>Switch to the <span>data state</span>. Emit the current DOCTYPE token.</dd>
 
    <dt>U+0022 QUOTATION MARK (&quot;)</dt>
-   <dd><span>Parse error</span>. Set the DOCTYPE token's system identifier to the empty string (not
+   <dd>This is a <dfn data-x="parse-error-missing-whitespace-after-doctype-public-identifier">missing-whitespace-after-doctype-public-identifier</dfn>
```
Comment (inikulin) [inline source]: Should be same as below: `missing-whitespace-between-doctype-public-and-system-identifiers`. I guess it's caused by my misleading comment again.

File: source [hunk 5]
```diff
@@ -103264,19 +103278,23 @@ dictionary <dfn>StorageEventInit</dfn> : <span>EventInit</span> {
...
    <dt>U+0027 APOSTROPHE (')</dt>
-   <dd><span>Parse error</span>. Set the DOCTYPE token's system identifier to the empty string (not
+   <dd>This is a <dfn data-x="parse-error-missing-whitespace-between-doctype-public-and-system-identifiers">missing-whitespace-between-doctype-public-and-system-identifiers</dfn>
+   <span>parse error</span>. Set the DOCTYPE token's system identifier to the empty string (not
    missing), then switch to the <span>DOCTYPE system identifier (single-quoted) state</span>.</dd>
 
    <dt>EOF</dt>
-   <dd><span>Parse error</span>. Set the DOCTYPE token's <i data-x="force-quirks flag">force-quirks
-   flag</i> to <i>on</i>. Emit that DOCTYPE token. Emit an end-of-file token.</dd>
+   <dd>This is an <span data-x="parse-error-eof-in-doctype">eof-in-doctype</span><span>parse error</span>.
```
Comment (inikulin) [inline source]: Missing space before `parse error`.

File: source [hunk 6]
```diff
@@ -103307,11 +103325,13 @@ dictionary <dfn>StorageEventInit</dfn> : <span>EventInit</span> {
    the <span>DOCTYPE system identifier (single-quoted) state</span>.</dd>
 
    <dt>EOF</dt>
-   <dd><span>Parse error</span>. Set the DOCTYPE token's <i data-x="force-quirks flag">force-quirks
+   <dd>This is an <span data-x="parse-error-eof-in-doctype">eof-in-doctype</span><span>parse error</span>.
```
Comment (inikulin) [inline source]: Missing space before `parse error`.

File: source [hunk 7]
```diff
@@ -103375,15 +103400,18 @@ dictionary <dfn>StorageEventInit</dfn> : <span>EventInit</span> {
    the <span>DOCTYPE system identifier (single-quoted) state</span>.</dd>
 
    <dt>U+003E GREATER-THAN SIGN (&gt;)</dt>
-   <dd><span>Parse error</span>. Set the DOCTYPE token's <i data-x="force-quirks flag">force-quirks flag</i> to <i>on</i>.
+   <dd>This is an <span data-x="parse-error-abrupt-closing-of-doctype">abrupt-closing-of-doctype</span>.
```
Comment (inikulin) [inline source]: Missing `<span>parse error</span>`

**Review by inikulin — CHANGES_REQUESTED** [review]

**Comment by diervo** [comment]
//cc @inikulin, once you approve, I will do the minor changes in the other two repos

**Review by inikulin — CHANGES_REQUESTED** [review]
@diervo a couple of issue left that were not fixed:
https://github.com/HTMLParseErrorWG/html/pull/12#discussion_r114756584
https://github.com/HTMLParseErrorWG/html/pull/12#discussion_r114756846

**Comment by diervo** [comment]
@inikulin Done, sorry for being sloppy

**Comment by inikulin** [comment]
@diervo No worries, sorry for the forced hurry.

**Comment by diervo** [comment]
@inikulin No this is great! If we could merge everything by the end of the week will be awesome. The sooner we finish the early I can do some integrations we are blocked on :)

**Comment by diervo** [comment]
Once you merge this, I will take care of the other two repos, then is only one left.

**Comment by inikulin** [comment]
@diervo Give me an hour, I'm in the middle of rebasing our work on top of the upstream changes.

**Comment by diervo** [comment]
No rush, I have a couple of hours in the afternoon to close whatever is remaining

**Comment by inikulin** [comment]
@diervo This one still not fixed: https://github.com/HTMLParseErrorWG/html/pull/12#discussion_r114756584

File: source (hunk 6 above)
Comment (inikulin) [reply source]: This one still not fixed. Missing space before `<span>parse error</span>` here.

**Review by inikulin — CHANGES_REQUESTED** [review]

**Comment by diervo** [comment]
@inikulin done

**Review by inikulin — APPROVED** [review]

**Pull request merged**